### PR TITLE
task 7

### PR DESCRIPTION
**Pull request by Bokbulbok7**

### CODE REVIEW

File: db/models/commentsModel.js [hunk 1]
```diff
@@ -10,3 +12,25 @@ exports.selectCommentsByArticleId = (articleId) => {
...
+exports.insertCommentByArticleId = (articleId, newComment) => {
+  const { username, body } = newComment;
+  if (!username || !body)
+    return Promise.reject({
+      status: 400,
+      msg: "Input should have username and body.",
+    });
+  return checkArticleExists(articleId)
+    .then(() => {
+      return checkUserExists(username);
```
Comment (nc-poppy-m) [inline db/models/commentsModel.js:25]: This 'checkUserExists' arguably should be handled with PSQL error handling, but this isn't mandatory

File: db/error.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ exports.handleCustomErrors = (err, req, res, next) => {
 };
 
 exports.handlePsqlErrors = (err, req, res, next) => {
-  if (err.code === "22P02") {
+  if (err.code === "22P02" || err.coded === "23503" || err.coded === "23502") {
```
Comment (nc-poppy-m) [inline db/error.js]: With the typo, does this work?

File: db/controllers/articlesController.js [hunk 3]
```diff
@@ -36,3 +40,33 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
+  const { articleId } = req.params;
+  const newComment = req.body;
+  return insertCommentByArticleId(articleId, newComment)
+    .then((comment) => {
+      return res.status(201).send({ comment });
+    })
+    .catch(next);
+};
+
+// exports.postCommentByArticleId = (req, res, next) => {
```
Comment (nc-poppy-m) [inline db/controllers/articlesController.js]: Make sure to remove commented out code before submission

**Review by nc-poppy-m — CHANGES_REQUESTED** [review]
Nearly there! A few changes to make before resubmission:
- If an extra property is passed, we want to receive a 201 and a created comment ignoring the additional properties
- What should happen if we have a non-existent but valid article_id?
- Does your PSQL error handler help you here? (also with the typo, have you checked that it works?) Refactoring to use this instead of your custom error handler for the checkUsername is optional, but definitely ensure it's still working.
- I can't find your endpoints.json here, but ensure it's committed and updated in your resubmission
- Remove commented out code in controller
Once this is done, resubmit this and we'll take another look at it :)

**Review by nc-mick-fay — APPROVED** [review]
Great additional tests and updating of your JSON file - you can merge into main when you're ready

**Pull request merged**